src/helper.h

```diff
@@ -31,6 +31,12 @@ struct get_json_ret_int {
   int v;
 };
 
+struct get_json_ret_array {
+  operator bool() const;
```

Comment:
On the subject of being more explicit, I don't like the name `helper` filename. If the filename is generic because there are helper functions for different things in here then consider splitting them into multiple files. Alternatively, if they are not needed by multiple classes they could go in the file and namespace of the class they help.